sys/include/net/nanocoap.h

```diff
@@ -1027,6 +1027,27 @@ ssize_t coap_opt_add_uquery2(coap_pkt_t *pkt, const char *key, size_t key_len,
  */
 ssize_t coap_opt_add_proxy_uri(coap_pkt_t *pkt, const char *uri);
 
+/**
+ * @brief   Encode the given array of characters as option(s) into pkt
+ *
+ * Use separator to split array of characters into multiple options.
+ *
+ * @post pkt.payload advanced to first byte after option(s)
+ * @post pkt.payload_len reduced by option(s) length
+ *
+ * @param[in,out] pkt         pkt referencing target buffer
+ * @param[in]     optnum      option number to use
+ * @param[in]     chars       array of characters to encode as option
+ * @param[in]     chars_len   length of @p chars
+ * @param[in]     separator   character used in @p string to separate parts
+ *
+ * @return        number of bytes written to buffer
+ * @return        <0 on error
+ * @return        -ENOSPC if no available options or insufficient buffer space
+ */
+ssize_t coap_opt_add_chars(coap_pkt_t *pkt, uint16_t optnum, const char *chars,
+                           size_t chars_len, char separator);
```

Comment:
What about naming the function `coap_opt_add_string_buf()` or `coap_opt_add_string2()`? There are other functions with a naming convention to use the data type as defined in RFC 7252, like `coap_opt_add_uint()` and `coap_opt_add_opaque()`. The current name is nice because it is short, but the function really does the same thing as `coap_opt_add_string()`, but just with a length parameter. It would be nice to keep a large function namespace easy to remember.

Reply:
I'd add `coap_opt_add_opaque_split()` into the suggestions, given that it's just a shortcut for multiple invocations of `coap_opt_add_opaque()` in a memchr-loop.

Reply:
> I'd add `coap_opt_add_opaque_split()` into the suggestions

At least in my mind, the intention is to add option(s) with a string data type, not opaque. Yes, the implementation of `coap_opt_add_chars()` uses `_add_opt_pkt()`. But ultimately _all_ option additions use this function as the final step to format the option in the PDU.

Reply:
I don't think it's worth blocking this PR on the name. At the very least the current name is short. We always can add an alias name in the form `coap_opt_add_string...()` for consistency later. I'll mark this resolved.